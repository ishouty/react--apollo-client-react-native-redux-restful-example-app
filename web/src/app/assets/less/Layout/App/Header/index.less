@headerHeight: 40px;

.header-container {
  width: 100%;
  height: @headerHeight;
  background-color: @color-white;
  border-bottom: 1px solid @color-light-gray;
  position: fixed;
  z-index: 10;
  top: 0px;

  i {
    font-size: 1.5em;
  }

  .header-body {
    margin: 0 auto;
  }

  /* Desktop */
  @media only screen and (min-width: 1012px) {


  }

  /* Tablets */
  @media only screen and (max-width: 768px) {

    span {
      display : none;
    }

  }

}
