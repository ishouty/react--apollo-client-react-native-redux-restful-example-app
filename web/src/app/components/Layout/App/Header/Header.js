import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Row, Col } from 'react-bootstrap'
import PropTypes from 'prop-types'
//Components
import Navigation from './../../Navigation/Navigation'

@connect((store) => {
  return {
    user: store.user,
    navigation: store.navigation.navigationLinks.links
  }
})
class Header extends Component {
  render() {
    const { navigation } = this.props

    return (
      <div className="header-container">
        <Row className="header-body">
          <Navigation navigationLinks={navigation} />
        </Row>
      </div>
    )
  }
}

export default Header
