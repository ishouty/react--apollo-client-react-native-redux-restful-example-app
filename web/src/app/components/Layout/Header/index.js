import React, { Component } from 'react'
import { connect } from 'react-redux'
import ProfileAvatarName from '../../ProfileAvatarName/ProfileAvatarName'
import PropTypes from 'prop-types'

@connect((store) => {
  return {
    profile: store.profile
  }
})
class Header extends Component {
  render() {
    const { profile } = this.props

    return (
      <ProfileAvatarName profile={profile.profile} size="medium" />
    )
  }
}

Header.propTypes = {
  profile: PropTypes.object
}

export default Header
