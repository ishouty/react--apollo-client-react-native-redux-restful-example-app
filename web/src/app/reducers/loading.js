import * as actionTypes from '../actions/actionTypes'
const initialState = {
  loading: {
    common: false
  }
}

/**
 * Reducer deals with all the functionality of the loading bar
 * @param state
 * @param action
 * @returns {Array}
 */
export default function reducer(state = initialState, action = {}) {
  const { type, payload } = action

  switch (type) {
    case actionTypes.LOADING_BTN:
    case actionTypes.LOADING_PAGE:
      return {
        ...state,
        loading: payload
      }

    default:
      return state
  }
}
