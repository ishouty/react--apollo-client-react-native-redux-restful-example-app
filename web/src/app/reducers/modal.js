import * as actionTypes from "../actions/actionTypes"

export const initialState = {
    modal: {
        userDetails: {
            display: false,
            details: {
                name: {
                    first: '',
                    last: ''
                },
                gender: '',
                email: '',
                phone: '',
                picture: '',
                location: {
                    street: '',
                    city: '',
                    postcode: ''
                }
            }
        }
    }
}

export default function reducer(state = initialState, action = {}) {
    const {type, payload} = action

    switch (type) {

        case actionTypes.USER_DETAIL_MODAL:

            return {
                ...state,

                modal: {

                    userDetails: {
                        display: payload.display,
                        details: payload
                    }
                }
            }


        case actionTypes.CLEAR_ALL_MODAL:
        default:
            return initialState


    }

}
