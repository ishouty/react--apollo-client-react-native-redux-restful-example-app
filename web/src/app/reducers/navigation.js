import * as actionTypes from "../actions/actionTypes"
import { NavigationLinkText } from "../constants/text/text"

export const initialState = {
    previousRoute: [],
    navigationLinks: {
        links: [
            {
                title: NavigationLinkText.item1,
                href: "/app/users",
                className: "nav-users",
                icon: "fa-user-circle-o"
            },
            {
                title: NavigationLinkText.item2,
                href: "/app/female-users",
                className: "nav-female-users",
                icon: "fa-female"
            },
            {
                title: NavigationLinkText.item3,
                href: "/app/male-users",
                className: "nav-male-users",
                icon: "fa-male"
            },
            {
                title: NavigationLinkText.item4,
                href: "/sign-out",
                className: "nav-sign-out",
                icon: "fa-sign-out"
            }
        ]
    }
}

export default function reducer(state = initialState, action = {}) {
    const { type, payload } = action

    switch (type) {

        case actionTypes.SAVE_HISTORY:

            var items = state.previousRoute
            items.push(payload)

            return {
                ...state,
                previousRoute: items
            }

        default:
            return state
    }
}
