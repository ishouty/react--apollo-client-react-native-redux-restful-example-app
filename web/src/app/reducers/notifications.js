import * as actionTypes from '../actions/actionTypes'

const initialState = {
    banner: {
        fail: false,
        success: false,
        message: ''
    }
}

export default function reducer(state = initialState, action = {}) {
    const { type, payload } = action

    switch (type) {

        case actionTypes.BANNER_NOTIFICATION:
            return {
                banner: payload
            }

        case actionTypes.CLEAR_STATE:
            return initialState

        default:
            return state

    }
}