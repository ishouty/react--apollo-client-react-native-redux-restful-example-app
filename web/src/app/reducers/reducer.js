import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { syncHistoryWithStore, routerReducer } from 'react-router-redux'

import loading from './loading'
import navigation from './navigation'
import users from './users'
import notifications from './notifications'
import modal from './modal'

export default combineReducers({
  form,
  navigation,
  loading,
  users,
  notifications,
  modal,
  routing: routerReducer
})
