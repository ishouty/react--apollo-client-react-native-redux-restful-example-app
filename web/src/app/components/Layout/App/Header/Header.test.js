import React from "react"
import {shallow} from "enzyme"
import toJson from "enzyme-to-json"
import Header from "./Header"
import { initialState } from  "../../../../reducers/navigation"

describe('Header', ()=> {

    test('renders component and match snap shot', () => {

        const component = shallow(<Header navigation={initialState.navigationLinks.links} />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()

    })

})