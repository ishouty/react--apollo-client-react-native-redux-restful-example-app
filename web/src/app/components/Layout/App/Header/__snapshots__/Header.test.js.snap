// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders component and match snap shot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
